rust/parquet/src/arrow/arrow_writer.rs

```diff
@@ -665,6 +670,16 @@ where
     values
 }
 
+fn get_bool_array_slice(array: &arrow_array::BooleanArray) -> Vec<bool> {
+    let mut values = Vec::with_capacity(array.len() - array.null_count());
```

Comment:
The logic that we were using was incorrect (at least once we get to lists). #8792 fixes this, so if it gets merged first, we would need to change this to populate values correctly .